src/ray/util/event.h

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <boost/asio/ip/host_name.hpp>
+#include <cmath>
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <vector>
+#include "ray/util/logging.h"
+#include "ray/util/util.h"
+#include "src/ray/protobuf/event.pb.h"
+
+namespace ray {
+
+#define RAY_EVENT(event_type, label) \
+  ::ray::RayEvent(::ray::rpc::Event_Severity::Event_Severity_##event_type, label)
+
+class LogBasedEventReporter {
```

Comment:
This class doesn't seem to have functions in .cc. Is it intended?

Reply:
This is the interface of our EventReporter. Our log file reporter implementation will inherit this interface, and the specific implementation will be proposed in the next PR

Reply:
Then aren't we supposed to change the class name? The class name looks like the implementation.

Reply:
Maybe Just name it BasedEventReporter ?